testtools/files.go

```diff
@@ -129,3 +139,232 @@ func CheckFiles(t *testing.T, dir string, files []FileSpec) {
 		}
 	}
 }
+
+type TestGazelleGenerationArgs struct {
+	// name is the name of the test.
+	Name string
+	// testDataPath is the workspace relative path to the test data directory.
+	TestDataPath string
+	// gazelleBinaryDir is the workspace relative path to the location of the gazelle binary
+	// we want to test.
+	GazelleBinaryDir string
+	// gazelleBinaryName is the name of the gazelle binary target that we want to test.
+	GazelleBinaryName string
+
+	// The suffix for all test input build files. Includes the ".".
+	// Default: ".in", so input BUILD files should be named BUILD.in.
+	BuildInSuffix string
+
+	// The suffix for all test output build files. Includes the ".".
+	// Default: ".out", so out BUILD files should be named BUILD.out.
+	BuildOutSuffix string
+}
+
+func NewTestGazelleGenerationArgs() *TestGazelleGenerationArgs {
+	return &TestGazelleGenerationArgs{
+		BuildInSuffix:  ".in",
+		BuildOutSuffix: ".out",
+	}
+}
+
+// TestGazelleGenerationOnPath runs a full gazelle binary on a testdata directory.
+// With a test data directory of the form:
+//└── <testDataPath>
+//    └── some_test
+//        ├── WORKSPACE
+//        ├── README.md --> README describing what the test does.
+//        ├── test.yaml --> YAML file for test configuration.
```

Comment:
The YAML file feels kind of unnecessary to me. WDYT about making a file called `stderr` and `stdout` describing the outputs and passing the `$(location)` of those in as arguments to the `gazelle_generation_test` rule? You can also pass in the `exit_code` that way and that completely eliminates the need to do YAML un/marshalling in this code.

Reply:
I guess the issue is that we need a different one per test? Could we just look for files named that in each test directory when walking the tree?